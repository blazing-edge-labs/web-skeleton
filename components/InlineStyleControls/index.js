import React from 'react'
import PropTypes from 'prop-types'
import { INLINE_STYLES } from 'constants/richTextEditor'
import StyleButton from 'components/StyleButton'

const InlineStyleControls = (props) => {
  const { editorState, onToggle } = props
  const currentStyle = editorState.getCurrentInlineStyle()

  return (
    <div>
      {INLINE_STYLES.map(type => (
        <StyleButton
          key={type.label}
          active={currentStyle && currentStyle.has(type.style)}
          label={type.label}
          onToggle={onToggle}
          style={type.style}
        />
      ))}
    </div>
  )
}

InlineStyleControls.propTypes = {
  editorState: PropTypes.object.isRequired,
  onToggle: PropTypes.func.isRequired,
}

export default InlineStyleControls
