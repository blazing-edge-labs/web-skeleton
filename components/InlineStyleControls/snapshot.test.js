import React from 'react'
import renderer from 'react-test-renderer'
import InlineStyleControls from './'

describe('InlineStyleControls component snapshot', () => {
  it('renders basic required data', () => {
    const tree = renderer.create(
      <InlineStyleControls
        editorState={{
          getCurrentInlineStyle: () => null,
        }}
        onToggle={() => {}}
      />
    )

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
