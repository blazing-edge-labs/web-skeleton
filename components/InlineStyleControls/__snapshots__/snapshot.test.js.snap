exports[`InlineStyleControls component snapshot renders basic required data 1`] = `
<div>
  <span
    onMouseDown={[Function]}>
    Bold
  </span>
  <span
    onMouseDown={[Function]}>
    Italic
  </span>
  <span
    onMouseDown={[Function]}>
    Underline
  </span>
  <span
    onMouseDown={[Function]}>
    Monospace
  </span>
</div>
`;
